lib/iris/tests/unit/fileformats/netcdf/test_Saver.py

```diff
@@ -612,5 +612,162 @@ def test_laea_cs(self):
+        cubes = iris.load(file_in)
+        # ensure cube order is the same:
+        cubes.sort(key=lambda cube: cube.cell_methods[0].method)
+        dtype = 'i2'
+        # Write Cube to netCDF file.
+        with self.temp_filename(suffix='.nc') as file_out:
+            iris.save(cubes, file_out, packing=dtype)
+            # Check the netCDF file against CDL expected output.
+            self.assertCDL(file_out, ('unit', 'fileformats', 'netcdf',
+                                      'save',
+                                      'multi_packed_single_dtype.cdl'))
+            packedcubes = iris.load(file_out)
+            packedcubes.sort(key=lambda cube: cube.cell_methods[0].method)
+            for cube, packedcube in zip(cubes, packedcubes):
+                sf, ao = self._get_scale_factor_add_offset(cube, dtype)
+                decimal = int(-np.log10(sf))
+                # Check that packed cube is accurate to expected precision
+                self.assertArrayAlmostEqual(cube.data, packedcube.data,
+                                            decimal=decimal)
+
+    def test_multi_packed_multi_dtype(self):
+        """Test saving multiple packed cubes with pack_dtype list."""
+        # Read PP input file.
+        file_in = tests.get_data_path(('PP', 'cf_processing',
+                                       'abcza_pa19591997_daily_29.b.pp'))
+        cubes = iris.load(file_in)
+        # ensure cube order is the same:
+        cubes.sort(key=lambda cube: cube.cell_methods[0].method)
+        datatype = 'i2'
+        scale_factor, offset = self._get_scale_factor_add_offset(cubes[0],
+                                                                 datatype)
+        packdict = dict(dtype=datatype, scale_factor=scale_factor,
+                        add_offset=offset)
+
+        dtypes = [packdict, None, 'u2']
+
+        # Write Cube to netCDF file.
+        with self.temp_filename(suffix='.nc') as file_out:
+            iris.save(cubes, file_out, packing=dtypes)
+            # Check the netCDF file against CDL expected output.
+            self.assertCDL(file_out, ('unit', 'fileformats', 'netcdf',
+                                      'save',
+                                      'multi_packed_multi_dtype.cdl'))
+            packedcubes = iris.load(file_out)
+            packedcubes.sort(key=lambda cube: cube.cell_methods[0].method)
+            for cube, packedcube, dtype in zip(cubes, packedcubes, dtypes):
+                if dtype:
+                    sf, ao = self._get_scale_factor_add_offset(cube, dtype)
+                    decimal = int(-np.log10(sf))
+                    # Check that packed cube is accurate to expected precision
+                    self.assertArrayAlmostEqual(cube.data, packedcube.data,
+                                                decimal=decimal)
+                else:
+                    self.assertArrayEqual(cube.data, packedcube.data)
+
```

Comment:
@jswanljung Awesome tests, thanks!

However, I'd consider them more as `integration` tests rather than `unit` tests ... @ajdawson and @marqh would you agree?

If so, then move them to `iris/tests/integration/test_netcdf.py` instead ...